src/python/WMCore/WMSpec/WMTask.py

```diff
@@ -331,6 +332,42 @@ def build(self, workingDir):
         master(self)
         return
 
+    def addEnvironmentVariables(self, envDict):
+        """
+        _addEnvironmentVariables_
+
+        add a key = value style setting to the environment for this
+        task
+        """
+        for key in envDict.keys():
+            setattr(self.data.environment, key, envDict[key])
+        for task in self.childTaskIterator():
+            task.addEnvironmentVariables(envDict)
+        return
+
+    def setStepOverrideCatalog(self, tfcFile):
+        """
+        _setStepOverrideCatalog_
+
+        Used for setting overrideCatalog option for each step in the task.
+        """
+        for step in self.steps().nodeIterator():
+            step = CMSSWStepHelper(step)
```

Comment:
I'm also not sure. Right now, this is adding `overrideCatalog` to every step, regardless of step type. I'm just using `CMSSWStepHelper.setOverrideCatalog` just because it already existed. I did not pay attention to the specific step type. Maybe we should migrate `setOverrideCatalog` from `CMSSWStepHelper` to `CoreHelper`?

Reply:
Sorry, I just realize this only makes sense for CMSSW steps, as it is a CMSSW option. I'll implement this change and  test that everything is ok.

Reply:
Done. I tested it and works the same. Only that now it does not create unnecessary attributes for non-CMSSW steps.